0000-ldap-sync-tool.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: `mgr-ldapsync`: LDAP sync tool
+- Start Date: 2019, Oct 28
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The `mgr-ldapsync` is a tool that performs synchronisation of the
+users between Uyuni server and LDAP.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni server can authenticate users from LDAP via PAM. For this, users
+needs to be added to Uyuni with the flag "PAM authentication" and thus
+their credentials will be verified via LDAP directory. Synchronisation
+tool should transparently synchronise users from certain groups and/or
+roles in LDAP and reflect that in Uyuni server, overriding local
+changes.
+
+Often LDAP configuration at infrastructures is unpredictable. This
+tool should bring certain boundaries and over up most common LDAP
+setups for users and their roles management.
+
+# Detailed design
+[design]: #detailed-design
+
+The tool should work as an aggregator, collecting users from LDAP and
+Uyuni server synchronising them.
+
+## Functionality Aspects
+
+The following functions should be implemented:
+
+- Distinguish between local users in Uyuni and LDAP users.
+- Prevent UID clashes between the local users and LDAP users.
+- Prevent accidental modification or removal of the local user.
+- Flexible support various LDAP schemas, such as eDirectory, POSIX,
+  Active Directory etc.
+- Role management.
+- Dry run mode (preview only).
+
+## Configuration
+
+Configuration should be default in a separate file
+`/etc/rhn/ldapsync.conf`. Alternatively, user should be able to
+specify own path, in case the default one is not created or used.
+
+The main principle of the tool is to map specific roles and or groups
+in LDAP to the roles in the Uyuni server via the configuration. Adding
+or removing LDAP users from these groups or roles should reflect in
+Uyuni without additional manual changes in the Uyuni server.
+
+Configuration should be just a YAML file and the mapping should
+implement a relation between LDAP group to a set of roles from
+Uyuni. So each user in LDAP that is a member of such group would
+inherit these roles. For example:
+
+```yaml
+directory:
+  groups:
+    cn=sysop,ou=groups,dc=example,dc=com:
+      - activation_key_admin
+      - channel_admin
+      - config_admin
+```
+
+The configuration above would allow creating `groupOfNames` class in
+LDAP and add users via `member` attribute. As long as DN of that
+member is valid LDAP object and is not clashing with the local user in
+the Uyuni server, it should be synchronised. Similarly roles LDAP
+objects are configured as same as `groups` but via `roles`.
```

Comment:
It's not clear to me how this tool works. Does it create local Uyuni Server users for the LDAP imported users? Does it create organizations, groups, etc too? Or it just synchronizes user data for existing Uyuni Server users?

Reply:
It creates users in Uyuni, based on LDAP layout and assignes roles accordingly. All you change in LDAP reflects in Uyuni. Local Uyuni users (those aren't in LDAP and has no PAM auth marked) are ignored and should be managed locally using Uyuni UI.